wasm/jit/jit_amd64.go

```diff
@@ -1752,6 +1880,18 @@ func (c *amd64Compiler) setFunctionCallIndexFromConst(index int64) {
 	c.addInstruction(prog)
 }
 
+// ensureRegisterUnused ensures that a given register is not used by
+// any value. If the register is currently used, release the value
+// to the stack.
+func (c *amd64Compiler) ensureRegisterUnused(reg int16) {
```

Comment:
maybe this and then you can drop the doc?

```suggestion
func (c *amd64Compiler) onValueReleaseRegisterToStack(reg int16) {
```